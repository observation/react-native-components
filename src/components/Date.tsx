import React from 'react'
import { StyleProp, ViewStyle } from 'react-native'

import { Icon } from './Icon'
import IconText from './IconText'
import { theme } from '../styles'
import textStyle from '../styles/text'

type Props = {
  date: string
  containerStyle?: StyleProp<ViewStyle>
}

const Date = ({ date, containerStyle }: Props) => (
  <IconText
    icon={<Icon name="calendar-day" style={'solid'} color={theme.color.greySemi} size={theme.icon.size.medium} />}
    text={date}
    style={{
      containerStyle,
      textStyle: textStyle.light,
    }}
    singleLineText
  />
)

export default Date
