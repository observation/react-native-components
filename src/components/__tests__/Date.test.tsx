import React from 'react'

import { render } from '@testing-library/react-native'

import Date from '../Date'

describe('Date', () => {
  test('Rendering', () => {
    const { toJSON } = render(<Date date="4-10-2019 9:10" containerStyle={{ flex: 1 }} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
