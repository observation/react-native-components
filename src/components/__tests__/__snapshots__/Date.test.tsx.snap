// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Rendering 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "justifyContent": "center",
          "marginRight": 8,
        },
        undefined,
      ]
    }
  >
    <Icon
      color="#E6E6E6"
      name="calendar-day"
      size={14}
      type="solid"
    />
  </View>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "flexShrink": 1,
        },
        {
          "color": "#939393",
          "fontFamily": "Ubuntu",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": 20,
        },
      ]
    }
  >
    4-10-2019 9:10
  </Text>
</View>
`;
